docs/book/src/architecture/controllers/cluster.md

```diff
@@ -1 +1,59 @@
-# Cluster
+# Cluster Controller
+
+The Cluster controller's main responsibilities are:
+
+* Setting an OwnerReference on the infrastructure object referenced in `Cluster.Spec.InfrastructureRef`.
+* Cleanup of all owned objects so that nothing is dangling after deletion.
+* Keeping the Cluster's status in sync with the infrastructure Cluster's status.
+* Creating a kubeconfig secret for [target clusters](../reference/glossary#Workload cluster).
+
+## Contracts
+
+### Infrastructure Provider
+
+The general expectation of an infrastructure provider is to provision the necessary infrastructure components needed to
+run a Kubernetes cluster. As an example, the AWS infrastructure provider, specifically the AWSCluster reconciler, will
+provision a VPC, some security groups, an ELB, a bastion instance and some other components all with AWS best practices
+baked in. Once that infrastructure is provisioned and ready to be used the AWSMachine reconciler takes over and
+provisions EC2 instances that will become a Kubernetes cluster through some bootstrap mechanism.
+
+#### Requirements
+
+The InfrastructureCluster object **must** have a `status` object.
+
+The `status` object **must** have several fields defined:
+
+* `ready` - a boolean field that is true when the infrastructure is ready to be used.
+* `apiEndpoints` - a slice of strings that identifies each control plane node's apiserver endpoint or a slice with only
+one endpoint that is a load balancer for all control plane nodes.
```

Comment:
Should `ErrorMessage` and `ErrorReason` also be listed here?

Reply:
Ah indeed it should

Reply:
Upon further inspection these are not required, but I will add another section outlining how these are used.